pkg/util/provider/machinecontroller/machine.go

```diff
@@ -223,10 +236,13 @@ func (c *controller) deleteNodeToMachine(obj interface{}) {
 
 	machine, err := c.getMachineFromNode(key)
 	if err != nil {
-		klog.Errorf("Couldn't fetch machine %s, Error: %s", key, err)
-		return
-	} else if machine == nil {
-		return
+		if err.Error() == "Multiple machines matching node" {
```

Comment:
Why not use `multipleMachineMatchError` like above?

Reply:
Oh yes I forgot to make the change here I guess